packages/web/src/utils/third-party/cli-files/models/webpack-configs/browser.ts

```diff
@@ -120,30 +122,32 @@ export function getBrowserConfig(wco: WebpackConfigOptions) {
               }
             : {}),
           vendors: false,
-          // TODO(jack): Support both 4 and 5
           vendor: !!buildOptions.vendorChunk && {
             name: 'vendor',
             chunks: isWebpack5 ? (chunk) => chunk.name === 'main' : 'initial',
             enforce: true,
-            test: isWebpack5
-              ? /[\\/]node_modules[\\/]/
-              : (
-                  module: { nameForCondition?: Function },
-                  chunks: Array<{ name: string }>
-                ) => {
-                  const moduleName = module.nameForCondition
-                    ? module.nameForCondition()
-                    : '';
+            test: (
+              module: Module,
+              // `CacheGroupsContext`, but it's not exported from webpack types
+              context: { moduleGraph: ModuleGraph; chunkGraph: ChunkGraph }
+            ) => {
+              const moduleName = module.nameForCondition
+                ? module.nameForCondition()
+                : '';
+              // TODO(jack): Remove in Nx 13
+              const chunks = isWebpack5
+                ? context.chunkGraph.getModuleChunks(module)
```

Comment:
I was going for the same condition as the webpack 4.
I'm not very familiar with the webpack API, so I am not sure if there is a better way to achieve this. I found that there's the `chunksFilter` config option, but it's called only with the `chunk` value, so I'm not sure if it would work.